docs/introduction/friendly-url.md

```diff
@@ -0,0 +1,30 @@
+# Friendly URL
+
+Shopsys Framework comes with the implementation to support custom URLs for entities or other use you might need.
+By default, there are custom URLs implemented for `Product` detail, `Product` list, `Article` detail and `Brand` detail pages.
+Thanks to this functionality you can set your own URL or set of URLs to these entities.
+This functionality is provided by `FriendlyUrlDataProviderInterface` implementations, e.g. `\Shopsys\FrameworkBundle\Model\Product\ProductDetailFriendlyUrlDataProvider`.
+Such classes are automatically registered in `FriendlyUrlDataProviderRegistry`.
+The rest of the work is done automatically and URLs provided by these providers are now accessible via browser.
+
+## How to create new `FriendlyUrlDataProvider`
+
+- create new class with name ending with `FriendlyUrlDataProvider`
+- this class has to implement `FriendlyUrlDataProviderInterface`
+- this interface requires you to implement two methods *(see `ProductDetailFriendlyUrlDataProvider` class as an example of the implementation)*:
+    - `getFriendlyUrlData` method that generates `FriendlyUrlData` for all your entities
+    - `getRouteName` method returns name of route that you have to declare in `routing_friendly_url.yml` file like:
+        ```
+        front_<entity_name>_detail:
+            path: friendly-url
+            defaults: { _controller: ShopsysShopBundle:Front\<entity_name>:detail }
+        ```
+        this will route all URLs you have provided in `getFriendlyUrlData` method to `<entity_name>Controller::detailAction` method and provide you with ID of entity matching that URL
+        *(see [Symfony Documentation](https://symfony.com/doc/3.4/controller.html) to learn how to create new `Controller`)*
+- update your `<entity_name>Facade`:
+    - in `create` and `edit` method behind `$this->em->flush()` add similar code like this `$this->friendlyUrlFacade->createFriendlyUrls('front_<entity_name>_detail', $entity->getId(), $entity->getNames());`
```

Comment:
```suggestion
    - in `create` and `edit` methods, after calling `$this->em->flush()`, add similar code like this:
     ```
     $this->friendlyUrlFacade->createFriendlyUrls('front_<entity_name>_detail', $entity->getId(), $entity->getNames());
	```
```